clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/CompletableEventReaper.java

```diff
@@ -155,4 +156,13 @@ public int size() {
     public boolean contains(CompletableEvent<?> event) {
         return event != null && tracked.contains(event);
     }
+
+    public List<CompletableEvent<?>> uncompletedEvents() {
+        return tracked.stream()
+                .filter(e -> e instanceof CompletableApplicationEvent<?>)
+                .map(e -> (CompletableApplicationEvent<?>) e)
+                .filter(e -> !e.future().isDone())
+                .collect(Collectors.toList());
+    }
```

Comment:
You don't have to filter by `CompletableApplicationEvent` is because `tracked` holds `CompletableEvents` already. 

```suggestion
    public List<CompletableEvent<?>> uncompletedEvents() {
        return tracked.stream()
                .filter(e -> !e.future().isDone())
                .collect(Collectors.toList());
    }
```